src/Search/SearchHandler.php

```diff
@@ -39,14 +39,13 @@ class SearchHandler
      * NEXT_MAJOR: Change signature to __construct(bool $caseSensitive) and remove pool property.
      *
      * @param Pool|bool $deprecatedPoolOrCaseSensitive
-     * @param bool      $caseSensitive
      */
-    public function __construct($deprecatedPoolOrCaseSensitive, $caseSensitive = true)
+    public function __construct($deprecatedPoolOrCaseSensitive, bool $caseSensitive = true)
```

Comment:
We should remove the deprecated parameter before adding typehint. (And removing the deprecation trigger)

Is it clear to you how to do it ?

Reply:
You mean to reduce contstructor like this or something more/else is needed?

```
public function __construct( bool $caseSensitive = true)
{
    $this->caseSensitive = $caseSensitive;
}
```

Reply:
Yes